src/main/java/org/jasig/portlet/widget/links/ResourceLink.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.portlet.widget.links;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Data object for Resource Links.They have the following Attributes:
+ *  - title
+ *  - description
+ *  - icon
+ *  - url
+ *  - groups
+ */
+public class ResourceLink {
```

Comment:
:thought_balloon: :arrow_right: It would be nice in the future if Lombok's `@Data` could be used to auto generate classes like this.
Not a blocker for merging this PR though :white_check_mark: 

Ref: https://projectlombok.org/features/Data

Reply:
:1st_place_medal: If I have time, I will add it.